contrib/babelfishpg_common/src/varchar.c

```diff
@@ -1133,6 +1133,7 @@ bpchar(PG_FUNCTION_ARGS)
 		size_t		maxmblen;
 
 		maxmblen = pg_encoding_mbcliplen(collInfo.enc, tmp, byteLen, maxByteLen);
+		blankSpace = maxByteLen - maxmblen;
```

Comment:
Add some test cases to validate this changes; Have different collation at column level with different typmod and make sure that blank space are padded correctly especially for language such as Arabic

Reply:
Done